### PR TITLE
Turret fix fixes, fixes stationary AI, upload turrets not shooting heads.

### PR DESCRIPTION
**Pull request by MacBlaze1**
## About The Pull Request
close #46625 
Fixes very bad oversight, no need to revert my PR

## Why It's Good For The Game

oversight fixes are good for the game and for the health of the codebase, duh.

## Changelog
:cl:
fix: fixes AI upload and AI station turrets not shooting heads of staff or cyborgs.
/:cl:

### CODE REVIEW

**Comment by carshalash** [comment]
This still leaves turrets unable to shoot unsyched/emagged borgs.

**Comment by Detective-Google** [comment]
I'm so glad this guy completely ignored him when I said this would be a problem 👍

**Comment by FancyCyclops** [comment]
This is not a fix.

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -853,7 +854,8 @@
 	for(var/obj/machinery/porta_turret/T in control_area)
 		turrets |= T
 		T.cp = src
-
+		T.shoot_heads_of_staff = 1
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret.dm]: TRUE

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 2]
```diff
@@ -820,6 +820,7 @@
 	var/locked = TRUE
 	var/control_area = null //can be area name, path or nothing.
 	var/ailock = 0 // AI cannot use this
+	var/shoot_cyborgs = 0
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret.dm]: FALSE

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 3]
```diff
@@ -62,7 +62,7 @@
 	var/shoot_cyborgs = 0 // checks if it can shoot people that are cyborgs
 	var/shoot_heads_of_staff = 0 // checks if it can shoot at heads of staff
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret.dm]: FALSE

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 4]
```diff
@@ -62,7 +62,7 @@
 	var/shoot_cyborgs = 0 // checks if it can shoot people that are cyborgs
```
Comment (ShizCalev) [inline code/game/machinery/porta_turret/portable_turret.dm]: TRUE

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 2 above)
Comment (MacBlaze1) [reply code/game/machinery/porta_turret/portable_turret.dm]: Its an option now, heads can turn it on for AI uploads and shit, Im confused on what you want me to do, turn the option on from the start? Allow silicons to have the option? Probably just the AI honestly. Ill do that.

**Comment by MacBlaze1** [comment]
> This is not a fix.
> firefox_L9vaQWhvsF

AHHHHHHHHHHHHHH IT IS, Fucking read. It fixes an oversight I made. Heads can be shot at now, and are not treated differently by AI upload turrets or Sat turrets, Silicons I believe Ill have to set it as an option that the AI and humans can toggle on or off.

**Comment by MacBlaze1** [comment]
> I'm so glad this guy completely ignored me when I said this would be a problem 👍

I literally dont remember you saying this, I looked up discord results from you with "turrets", "ai upload" "ai" "sat" and nothing. You never brought this concern.

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 2 above)
Comment (ninjanomnom) [reply code/game/machinery/porta_turret/portable_turret.dm]: `FALSE` is a define used for bools, don't use `1` or `0`; use `TRUE` or `FALSE`

Comment (MacBlaze1) [reply code/game/machinery/porta_turret/portable_turret.dm]: Ahhh okay.

Comment (MacBlaze1) [reply code/game/machinery/porta_turret/portable_turret.dm]: I will admit Im not the one who set these booleans to 1 or 0.

Comment (ExcessiveUseOfCobblestone) [reply code/game/machinery/porta_turret/portable_turret.dm]: T/F didn't exist whenthese were added

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 5]
```diff
@@ -957,10 +959,15 @@
 	add_hiddenprint(user)
 	log_combat(user, src, "[enabled ? "enabled" : "disabled"]")
 	updateTurrets()
-
+/obj/machinery/turretid/proc/shoot_silicons(mob/user)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/porta_turret/portable_turret.dm]: this trash is literally copypasta'd thrice, cringe

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 6]
```diff
@@ -957,10 +959,15 @@
...
-
+/obj/machinery/turretid/proc/shoot_silicons(mob/user)
+	shoot_cyborgs = !shoot_cyborgs
+	add_hiddenprint(user)
+	log_combat(user, src, "[shoot_cyborgs ? "Shooting Borgs" : "Not Shooting Borgs"]")
+	updateTurrets()
 /obj/machinery/turretid/proc/updateTurrets()
 	for (var/obj/machinery/porta_turret/aTurret in turrets)
 		aTurret.setState(enabled, lethal)
+		aTurret.shoot_cyborgs = shoot_cyborgs
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/porta_turret/portable_turret.dm]: you should change the setState proc to handle cyborgshooting too

Comment (MacBlaze1) [reply code/game/machinery/porta_turret/portable_turret.dm]: Dont think it works with that, might.

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 7]
```diff
@@ -592,15 +592,15 @@
...
 
-/obj/machinery/porta_turret/proc/setState(on, mode)
+/obj/machinery/porta_turret/proc/setState(on, mode, shoot_cyborgs)
 	if(controllock)
 		return
 	src.on = on
 	if(!on)
 		popDown()
 	src.mode = mode
 	power_change()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/porta_turret/portable_turret.dm]: flip 02/03

**Pull request merged**